libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/entitlements/FilesEntitlement.java

```diff
@@ -145,12 +148,14 @@ private static Mode parseMode(String mode) {
     }
 
     private static BaseDir parseBaseDir(String baseDir) {
-        if (baseDir.equals("config")) {
-            return BaseDir.CONFIG;
-        } else if (baseDir.equals("data")) {
-            return BaseDir.DATA;
-        }
-        throw new PolicyValidationException("invalid relative directory: " + baseDir + ", valid values: [config, data]");
+        return switch (baseDir) {
```

Comment:
This is nice IMO, but I think it'll fail backport (8.x is not at this Java lang level IIRC). Just a warning.

Reply:
8.17 is at a minimum of Java 17 (based on https://www.elastic.co/support/matrix), so would expect this to be safe for backports as well